controllers/job_mgr/job_mgr_controller.go

```diff
@@ -0,0 +1,173 @@
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads/finalizers,verbs=update
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=resourceflavors,verbs=get;list;watch
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloadpriorityclasses,verbs=get;list;watch
+
+type LMEvalJob struct {
+	workloadv1alpha1.LMEvalJob
+}
+
+func ControllerSetUp(mgr manager.Manager, ns, configmap string, recorder record.EventRecorder) error {
+	ctx := context.TODO()
+	if err := jobframework.SetupWorkloadOwnerIndex(ctx, mgr.GetFieldIndexer(), workloadv1alpha1.GroupVersion.WithKind("LMEvalJob")); err != nil {
+		return fmt.Errorf("workload indexer: %w", err)
+	}
+	lmes.Job_mgr_enabled = true
+	return jobframework.NewGenericReconcilerFactory(
+		func() jobframework.GenericJob { return &LMEvalJob{} },
+		func(b *builder.Builder, c client.Client) *builder.Builder {
+			return b.Named("LMEvalJobWorkload")
+		},
+	)(mgr.GetClient(), mgr.GetEventRecorderFor("kueue")).SetupWithManager(mgr)
+
+}
+
+// Object returns the job instance.
+func (job *LMEvalJob) Object() client.Object {
+	return &job.LMEvalJob
+}
+
+// IsSuspended returns whether the job is suspended or not.
+func (job *LMEvalJob) IsSuspended() bool {
+	return job.Spec.Suspend
+}
+
+func (job *LMEvalJob) Suspend() {
+	job.Spec.Suspend = true
+}
+
+// RunWithPodSetsInfo will inject the node affinity and podSet counts extracting from workload to job and unsuspend it.
+func (job *LMEvalJob) RunWithPodSetsInfo(podSetsInfo []podset.PodSetInfo) error {
+	job.Spec.Pod.Affinity = convertToAffinity(podSetsInfo)
+	job.Spec.Suspend = false
+	return nil
+}
+
+// RestorePodSetsInfo will restore the original node affinity and podSet counts of the job.
+// Returns whether any change was done.
+func (job *LMEvalJob) RestorePodSetsInfo(podSetsInfo []podset.PodSetInfo) bool {
+	job.Spec.Pod.Affinity = convertToAffinity(podSetsInfo)
+	return true
+}
+
+// Finished means whether the job is completed/failed or not,
+// condition represents the workload finished condition.
+
+func (job *LMEvalJob) Finished() (condition metav1.Condition, finished bool) {
+	// ToDos: What should be in the condition and how will it be used?
+	condition = metav1.Condition{
+		Type:    "Finished",
+		Status:  "False",
+		Reason:  string(job.Status.Reason),
+		Message: job.Status.Message,
+	}
+	finished = false
+
+	if job.Status.State == workloadv1alpha1.CompleteJobState {
+		condition.Status = "True"
+		finished = true
+		return
+	}
+	return
+}
+
+// PodSets will build workload podSets corresponding to the job.
+func (job *LMEvalJob) PodSets() []kueue.PodSet {
+	log := log.FromContext(context.TODO())
+	pod := lmes.CreatePod(lmes.Options, &job.LMEvalJob, log)
+	podSet := kueue.PodSet{
+		Name:     job.Status.PodName,
```

Comment:
`job.Status.PodName` is updated after the LMES controller creates the job pod. Is this function triggered after the job pod is created?

Reply:
I believe PodSet is created(for Workloads) before the Pod can created.